main/src/tokenizer/gpt2_tokenizer.rs

```diff
@@ -201,6 +201,17 @@ mod tests {
         .cloned()
         .collect();
 
+        let special_token_map = SpecialTokenMap {
+            unk_token: "<|endoftext|>".to_string(),
```

Comment:
1. If you derive `Default` trait you can avoid putting `None` for most tokens and something like:
```rust
SpecialTokenMap {
    unk_token: Some("<|endoftext|>".into()),
}
```
2. I think that we shouldn't use `SpecialTokenMap` for hardcoded values and we can do something similar to what I did in #76. But it's about aesthetics and last word is yours.

Reply:
Thank you - I agree the `Default` implementation allows for conciseness, however here I believe explicitly listing the special token values improves readability and allows seeing at a glance the default configuration for special tokens for a vocab (i.e. the fact that `eos` is `None` is valuable information.